### PR TITLE
Simplify the python bundle example

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR simplified the python based bundle example according to @wyli 's comments.
It removed the confusing "private" `@property`.

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

File: bundle/python_bundle_workflow/scripts/inference.py [hunk 1]
```diff
@@ -140,32 +134,27 @@ def _evaluator(self):
...
             StatsHandler(name="eval_log", output_transform=lambda x: None),
-            CheckpointLoader(load_path=self.bundle_root + "/models/model.pt", load_dict={"net": self._network_def}),
+            CheckpointLoader(load_path=self.bundle_root + "/models/model.pt", load_dict={"net": self.network_def}),
         ]
 
-    @property
     def _dataset(self):
         return Dataset(
-            data=self._dataset_data,
+            data=self.dataset_data,
```
Comment (vfdev-5) [inline bundle/python_bundle_workflow/scripts/inference.py:146]: Shouldn't this be `self._dataset_data` ?

Comment (Nic-Ma) [reply bundle/python_bundle_workflow/scripts/inference.py:146]: Hi @vfdev-5 ,

Here we call the `dataset_data` property, and the `_get_property()` function will call the `def _dataset_data()`.

Thanks.

File: bundle/python_bundle_workflow/scripts/train.py [hunk 2]
```diff
@@ -110,27 +110,23 @@ def _get_property(self, name, property):
             self._props[name] = value
         else:
             try:
-                value = self.__getattribute__(f"_{name}")
+                value = getattr(self, f"get_{name}")()
             except AttributeError:
                 if property[BundleProperty.REQUIRED]:
                     raise ValueError(f"unsupported property '{name}' is required in the bundle properties.")
```
Comment (wyli) [inline bundle/python_bundle_workflow/scripts/train.py:116]: perhaps this error message can provide more info about how to resolve the error? like adding the property? create `get_{name}` method?

Comment (Nic-Ma) [reply bundle/python_bundle_workflow/scripts/train.py]: Sure, added in the latest commit.

Thanks.

File: bundle/python_bundle_workflow/scripts/inference.py [hunk 3]
```diff
@@ -98,7 +98,10 @@ def _get_property(self, name, property):
                 value = getattr(self, f"get_{name}")()
             except AttributeError:
                 if property[BundleProperty.REQUIRED]:
-                    raise ValueError(f"unsupported property '{name}' is required in the bundle properties.")
+                    raise ValueError(
+                        f"unsupported property '{name}' is required in the bundle properties,"
```
Comment (vfdev-5) [inline bundle/python_bundle_workflow/scripts/inference.py:102]: Nit: the word property we can replace with attribute as you removed properties ?

Comment (Nic-Ma) [reply bundle/python_bundle_workflow/scripts/inference.py:102]: Hi @vfdev-5 ,

I see your point.
But we are using `property` in the context:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/properties.py#L30

Thanks.

**Pull request merged**